### PR TITLE
Send config file diagnonstics event even when no errors were found

### LINKED ISSUE #10473: TSServer: config file diagnostics event not sent if config file changes
**TypeScript Version:**  2.0.0
**Code**
_tsconfig.json_

``` json
{
    "compilerOptions": {
        "module": "commonjs",
        "aaa": 10
    }
}
```

The server sends and event 

```
Event received: configFileDiag (0).
Data: {
    "triggerFile": "p:/mseng/VSCode/Playgrounds/ts-2.0/test.ts",
    "configFile": "p:/mseng/VSCode/Playgrounds/ts-2.0/tsconfig.json",
    "diagnostics": [
        {
            "text": "Unknown compiler option 'aaa'."
        }
    ]
}
```

which is correct. Fix the error in the tsconfig.json and save.

Observe: no configFileDiag update is sent. We either need an event to reqeuest configFileDiag or it should automatically happen when the tsconfig.json file is saved and recognized by the server.

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #10473

### CODE REVIEW

**Comment by vladima** [comment]
can you add tests?

File: src/server/session.ts [hunk 1]
```diff
@@ -162,17 +164,20 @@ namespace ts.server {
...
             private hrtime: (start?: number[]) => number[],
             protected logger: Logger,
-            protected readonly canUseEvents: boolean) {
+            protected readonly canUseEvents: boolean,
+            eventHandler?: ProjectServiceEventHandler) {
 
-            const eventHandler: ProjectServiceEventHandler = canUseEvents
-                ? event => this.handleEvent(event)
+            this.eventHander = canUseEvents
+                ? eventHandler
```
Comment (vladima) [inline src/server/session.ts]: nit.
```ts
this.eventHandler = canUseEvent 
    ? eventHandler || (event => this.defaultEventHandler(event))
    : undefined
```

**Comment by zhengbli** [comment]
@vladima updated. Other comments?

**Comment by mhegazy** [comment]
@zhengbli can you port this to release-2.0.5 as well.

**Comment by zhengbli** [comment]
@mhegazy Done

**Pull request merged**